src/osd/ReplicatedBackend.cc

```diff
@@ -619,16 +619,14 @@ void ReplicatedBackend::submit_transaction(
   op_t->register_on_applied(
     parent->bless_context(
       new C_OSD_OnOpApplied(this, &op)));
-  op_t->register_on_applied(
-    new ObjectStore::C_DeleteTransaction(op_t));
   op_t->register_on_commit(
     parent->bless_context(
       new C_OSD_OnOpCommit(this, &op)));
-
-  list<ObjectStore::Transaction*> tls;
-  tls.push_back(op_t);
+  vector<ObjectStore::Transaction> tls;
+  tls.push_back(std::move(*op_t));
   parent->queue_transactions(tls, op.op);
   delete t;
+  delete op_t;
```

Comment:
@cbodley Good catch ! This  simple fix we can do now to get rid of one new/delete from io path..
Changed code , running 'make check' will update the pull request once succeeded..